fuzz_lightyear/generator.py

```diff
@@ -96,26 +105,63 @@ def _add_request_to_sequence(
     for sequence in seed:
         for tag_group in dir(client):
             for request in _generate_requests(tag_group):
-                # Attempt to add every single permutation first.
-                new_sequence = sequence + [request]
-
-                # TODO: sequence pruning. We would get a lot better results if
-                #       we only added a request that consumes resources that
-                #       have been created by the sequence so far. However, due
-                #       to the nature of microservices, we may need to rely on
-                #       factories for this.
-                #
-                #       This is further made complicated because we have no idea
-                #       whether a request should consume an existing resource,
-                #       use a factory, or fuzz a value.
-                #
-                #       This means, we need to do all permutations, because we
-                #       can't possibly know whether to eagerly prune.
-                output.append(new_sequence)
+                neighbor_requests = request_graph[request.operation_id]
+                # We have no idea whether a request should consume an existing
+                # resource, use a factory, or fuzz a value.
+                # However, we want to move towards building sequences of requests
+                # that only consume resources created by previous requests (the
+                # algorithm described in the original Microsoft paper)
+                # Therefore, add the request to the sequence if it shares an edge with
+                # ANY of the requests in the sequence- leaving room for factory
+                # usage, but still keeping in line with the spirit of RESTler.
```

Comment:
I wonder whether this is a sane assumption, given two cases:

1. The initial base case of sequence length == 1. There would be no edges it shares (because our search space is just one node?)

2. Endpoints that solely depend on factory usage (so no common edges)?

Reply:
I'm not 100% certain what assumption is being addressed here. I'm assuming (hehe) that it's 
> ...we want to move towards building sequences of requests that only consume resources created by previous requests

If so, (1) is addressed in lines 90-95- all sequences of length == 1 are attempted, and if any of them succeed, they are added to the list of `good_sequences` in `generate_sequences`. This will occur even if values within them are supplied by a factory, or fuzzed.

(2) is a little bit more delicate. For endpoints that require solely on factory usage, that share no edges with any other nodes/operations in the request_graph, the request will get executed in the execution of sequences of length==1 . After that occurs, does it matter if that request is part of (or not part of?) a sequence? If it is not connected to the rest of the `request_graph`, we can't necessarily make the assumption that it doesn't manipulate any shared resources, but we can make the assumption that nothing that the request returns or produces will be used in any subsequent request. I'm not sure how we would decide if an operation that has no edges would "belong" in a sequence.

Reply:
Ah whoops. I forgot about lines 90-95. My bad. This might make sense then.